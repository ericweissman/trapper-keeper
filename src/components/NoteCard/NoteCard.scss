a {
  color: inherit;
  text-decoration: none;
}

.card {
  border: 1px solid darkgrey;
  border-radius: 3px;
  margin: 20px;
  padding: 10px;
  background-color: $white;
  font-family: $font-secondary;
  background-color: $card-bg;
  box-shadow: 2px 2px 6px 0px rgba(0,0,0,1);
  &:hover {
    box-shadow: 2px 2px 6px 0px rgba($primary-dark, 1);
  }
  h2 {
    font-family: $font-primary;
    color: $primary-dark;
    overflow-wrap: break-word;
  }
}

.card-title {
  margin: 0 0 20px;
  text-align: center;
  border-radius: 4px;
  padding: 0 0 5px 5px;
}

.incomplete-items {
  margin: 10px 0;
  background-color: $highlight;
  border-radius: 4px;
}

.complete-items {
  border-radius: 4px;
  background-color: $primary-light;
}


