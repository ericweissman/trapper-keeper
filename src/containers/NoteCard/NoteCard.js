import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import CompleteItem from '../../components/CompleteItem/CompleteItem.js'
import IncompleteItem from '../../components/IncompleteItem/IncompleteItem.js'
import PropTypes from 'prop-types';

export class NoteCard extends Component {

  render() {
    const incomplete = this.props.noteItems.map((item) => {
      if (!item.isCompleted) {
        return <IncompleteItem key={item.id} description={item.description}/>
      } else return []
    })
    const complete = this.props.noteItems.map((item) => {
      if (item.isCompleted) {
        return <CompleteItem key={item.id} description={item.description} />
      } else return []
    })

    return (
      <Link to={`/notes/${this.props.id}`}>
        <div className='card'>
          <h2 className='card-title'>{this.props.title}</h2>
          <div className='incomplete-items'>
            {incomplete}
          </div>
          <div className='complete-items'>
            {complete}
          </div>
        </div>
      </Link>
    )
  }
}

NoteCard.propTypes = {
  id: PropTypes.number,
  noteItems: PropTypes.array,
  title: PropTypes.string,
}

NoteCard.defaultProps = {
  noteItems: [],
  title: '',
}

export default NoteCard