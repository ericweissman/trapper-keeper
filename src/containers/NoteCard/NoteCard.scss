

.card {
  border: 2px solid black;
  margin: 20px;
  padding: 10px;
  background-color: $primary;
}

.card-title {
  margin: 0 0 20px;
}

.complete-items {
  background-color: grey;
}