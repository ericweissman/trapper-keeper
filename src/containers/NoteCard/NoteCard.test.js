import React from 'react';
import NoteCard from './NoteCard';
import { shallow } from 'enzyme';

describe('NoteCard', () => {
  let wrapper

  it('should match the snapshot', () => {
    //setup
    let mockNoteItems = [{ description: 'Note one of Title one', noteID: 1, id: 1, isCompleted: true }, { description: 'Note two of Title one', noteID: 1, id: 2, isCompleted: false }]
    let mockTitle = 'Title One'

    wrapper = shallow(<NoteCard
      title={mockTitle}
      noteItems={mockNoteItems}
    />)
    //expectation
    expect(wrapper).toMatchSnapshot()

  })
})