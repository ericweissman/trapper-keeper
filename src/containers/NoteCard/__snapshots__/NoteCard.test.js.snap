// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard should match the snapshot 1`] = `
<Link
  replace={false}
  to="/notes/"
>
  <div
    className="card"
  >
    <h2
      className="card-title"
    >
      Title One
    </h2>
    <div
      className="incomplete-items"
    >
      <IncompleteItem
        description="Note two of Title one"
        key="2"
      />
    </div>
    <div
      className="complete-items"
    >
      <CompleteItem
        description="Note one of Title one"
        key="1"
      />
    </div>
  </div>
</Link>
`;
